src/Console/Configuration/ConfigResolver.php

```diff
@@ -0,0 +1,113 @@
+<?php
+
+namespace Lmc\Steward\Console\Configuration;
+
+use Symfony\Component\Console\Input\InputDefinition;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\OptionsResolver\Exception\InvalidOptionsException;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+/**
+ * Resolve current configuration values based on CLI input and config file
+ */
+class ConfigResolver
+{
+    /** @var OptionsResolver */
+    private $optionsResolver;
+    /** @var InputDefinition */
+    private $inputDefinition;
+
+    /**
+     * @param OptionsResolver $optionsResolver
+     * @param InputDefinition $inputDefinition
+     */
+    public function __construct(OptionsResolver $optionsResolver, InputDefinition $inputDefinition)
+    {
+        $configurator = new OptionsResolverConfigurator();
+        $this->optionsResolver = $configurator->configure($optionsResolver);
+
+        $this->inputDefinition = $inputDefinition;
+    }
+
+    public function resolve(InputInterface $input, array $configFileInputOptions)
+    {
+        $config = $this->optionsResolver->resolve($configFileInputOptions);
+
+        $config = $this->setupAndTestDirs($input, $config);
+
+        return $config;
+    }
+
+    /**
+     * Setup values of configured directories and ensure the directories are accessible.
+     *
+     * @param InputInterface $input
+     * @param array $config Original configuration
+     * @throws \RuntimeException Thrown when directory is not accessible
+     * @return array Updated configuration
+     */
+    protected function setupAndTestDirs(InputInterface $input, array $config)
+    {
+        $relevantDirOptionsForCommand = $this->findRelevantDirOptionsForCurrentCommand();
+
+        /** @var $relevantDirOptionsForCommand InputOption[] */
+        foreach ($relevantDirOptionsForCommand as $dirOption) {
+            $cliOption = $dirOption->getName();
+            $cliValue = $input->getOption($dirOption->getName());
+            $wasPassedCliValue = ($input->getParameterOption('--' . $cliOption) !== false);
+            $configFileOption = str_replace('-', '_', $cliOption);
+            $configFileValue = isset($config[$configFileOption]) ? $config[$configFileOption] : null;
+
+            if ($wasPassedCliValue) { // CLI value has priority when passed
+                $currentValue = $cliValue;
+            } elseif ($configFileValue !== null) {
+                $currentValue = $configFileValue;
+            } else { // In not passed and not in config file use the default CLI value
+                $currentValue = $cliValue;
+            }
+
+            $this->throwExceptionIfDirNotReadable($currentValue, $dirOption);
```

Comment:
why not to use assertion? or at least name this method `assert` instead of `throwExceptionIf` ?
and btw called method `throwExceptionIfDirNotReadable` should be defined right after first call